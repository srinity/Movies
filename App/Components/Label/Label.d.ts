import { Component } from 'react'
import {
  TouchableOpacityProps,
  StyleProp,
  TextStyle,
  Animated
} from 'react-native'

export interface LabelProps extends TouchableOpacityProps {
  /**
   * Text shown inside the label.
   *
   */
  title: string

  /**
   * Optional placeholders values array that will be passed to ContentManager to replace any placeholder in the localization.
   *
   * @default []
   */
  placeHolders?: string[]

  /**
   * Optional styling for the text of the label.
   *
   */
  textStyle?: StyleProp<TextStyle>

  /**
   * Optional accessibilityLabel for the text of the label.
   *
   */
  textAccessibilityLabel?: string

  /**
   * Specifies whether fonts should scale to respect Text Size accessibility settings.
   *
   */
  allowFontScaling?: boolean
}

export interface AnimatedLabelProps
  extends LabelProps,
    Animated.AnimatedComponent<any> {}

export default class Label extends Component<LabelProps> {}

export class AnimatedLabel extends Component<AnimatedLabelProps> {}
