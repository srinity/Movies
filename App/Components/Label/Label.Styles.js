import { StyleSheet } from 'react-native'

import { Colors, Fonts } from '../../Theme'

export default StyleSheet.create({
  textStyle: {
    color: Colors.blackColor,
    fontSize: Fonts.h5
  },
  containerStyle: {
    width: '100%',
    justifyContent: 'center',
    alignItems: 'center'
  }
})
