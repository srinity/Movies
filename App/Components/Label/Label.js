import React, { Component } from 'react'
import { TouchableOpacity, Text, ViewPropTypes, Animated } from 'react-native'

import PropTypes from 'prop-types'

import styles from './Label.Styles'

class Label extends Component {
  render() {
    const { title, onPress, style, textStyle, ...props } = this.props
    return (
      <TouchableOpacity
        onPress={onPress}
        style={[styles.containerStyle, style]}
        {...props}
      >
        <Animated.Text style={[styles.textStyle, textStyle]}>
          {title}
        </Animated.Text>
      </TouchableOpacity>
    )
  }
}

Label.propTypes = {
  onPress: PropTypes.func,
  title: PropTypes.string.isRequired,
  textStyle: Text.propTypes.style,
  style: ViewPropTypes.style
}

const AnimatedLabel = Animated.createAnimatedComponent(Label)

export { AnimatedLabel }
export default Label
