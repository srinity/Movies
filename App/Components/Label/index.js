import Label, { AnimatedLabel } from './Label'

export { AnimatedLabel }
export default Label
