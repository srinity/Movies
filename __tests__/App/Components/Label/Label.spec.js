import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import Label from '../../../../App/Components/Label/Label'

describe('Label Component', () => {
  it('should render correctly', () => {
    const label = renderer.create(<Label title="testTitle" />)
    expect(label).toMatchSnapshot()
  })

  it('should invoke onPress correctly', () => {
    const onPress = jest.fn()

    const label = shallow(<Label title="testTitle" onPress={onPress} />)
    label.simulate('press')
    expect(onPress).toHaveBeenCalled()
  })
})
