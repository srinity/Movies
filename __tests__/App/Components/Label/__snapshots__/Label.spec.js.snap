// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label Component should render correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#000000",
        "fontSize": 14,
      }
    }
  >
    testTitle
  </Text>
</View>
`;
